pkg/reconciler/ingress/resources/reference_grant.go

```diff
@@ -18,16 +18,17 @@ package resources
 
 import (
 	"context"
+	"fmt"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	netv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
 	"knative.dev/pkg/kmeta"
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
-// Grant the resource "to" access to the resource "from"
+// MakeReferenceGrant Grant the resource "to" access to the resource "from"
 func MakeReferenceGrant(_ context.Context, ing *netv1alpha1.Ingress, to, from metav1.PartialObjectMetadata) *gatewayv1beta1.ReferenceGrant {
-	name := to.Name
+	name := fmt.Sprintf("%s-%s", ing.Name, to.Name)
```

Comment:
I think how we formulate the name is still incorrect when the names are very long you'll get a conflict.

I think we should just use the helper function we have see: https://github.com/knative/pkg/blob/b7bbf4be5dbd9a8b8b60e06094bd8930cf241357/kmeta/names.go#L42

Suffix would need to be a combination of the secret name, gateway name and namespace

Reply:
Good idea. I adjusted the code and changed the test to match the long name scenario.